logs/http_request_properties.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 Board of Trustees of the University of Illinois
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import "net/http"
+
+// HTTPRequestProperties is an entity which contains the properties of an HTTP request
+type HTTPRequestProperties struct {
+	Method     string
+	Path       string
+	RemoteAddr string
+	UserAgent  string
+}
+
+// Match returns true if the provided http.Request matches the defined request properties
+func (h HTTPRequestProperties) Match(r *http.Request) bool {
+	if h.Method != "" && h.Method != r.Method {
+		return false
+	}
+
+	if h.Path != "" && h.Path != r.URL.Path {
+		return false
+	}
+
+	if h.RemoteAddr != "" && h.RemoteAddr != r.RemoteAddr {
+		return false
+	}
+
+	if h.UserAgent != "" && h.UserAgent != r.UserAgent() {
+		return false
+	}
+
+	return true
```

Comment:
This will work, but I want to make sure that it is intentional that `HTTPRequestProperties` with all its attributes being empty should match any `http.Request`. It may be worth deciding on whether `nil`, `"all"`, or `""` should mean "all" across the system for consistency at some point as well.

Reply:
Yeah this is a good point. In this case I think this is the expected behavior since "" is teh default value and we are essentially wanting to ignore anything that is not specified. It would be nice to standardize this a bit as you described, however I think we will still run into situations where we have to make decisions on a case-by-case basis.